Sources/Classes/Headers/Public/RSDeviceModeManager.h

```diff
@@ -20,12 +20,14 @@
     RSEventFilteringPlugin *eventFilteringPlugin;
     RSConsentFilterHandler *consentFilterHandler;
     NSMutableDictionary<NSString*, id<RSIntegration>>* integrationOperationMap;
-    NSMutableDictionary<NSNumber*, RSMessage*> *eventReplayMessage;
     NSDictionary<NSString*, NSString*>* destinationsWithTransformationsEnabled;
     BOOL areFactoriesInitialized;
+    long beforeSDKInitEventTimestamp;
```

Comment:
If `beforeSDKInitEventTimestamp` is used to indicate the timestamp when the SDK was initialized previously, can we please rename it to something like `previousSDKInitTimeStamp` for better clarity?

Reply:
Same for all the methods dealing with `beforeSDKInitEventTimestamp`

Reply:
`beforeSDKInitEventTimestamp` means to denote the timestamp of the first event made before SDK initialisation is successful.

Here is the use case:
Suppose, someone updates the SDK from the version where the issue is present to the latest version containing the fix, and while SDK is being initialised, some events are made, and as of now SDK is yet to be initialised, events are stored in the DB. Now, suppose SDK is initialised properly and we are ready to mark previous SDK version events as device_mode_processing_done, we need to filter out the events which are made post-update, right? This variable helps in achieving that.

More details regarding this property:

- It will only be set once post-SDK updates.
- It is required to filter out the events made before SDK init is successful.

Reply:
Based on the discussion, we are going to completely get rid of this `timestamp` logic and we will basically refactor the past unprocessed event deletion logic in the EventRepository `init` method.